lib/portage/tests/resolver/test_merge_order.py

```diff
@@ -382,10 +382,12 @@ def testMergeOrder(self):
                 ambiguous_merge_order=True,
                 # The following merge order assertion reflects optimal order for
                 # a circular relationship which is DEPEND in one direction and
-                # RDEPEND in the other.
-                merge_order_assertions=(
-                    ("app-misc/circ-buildtime-a-1", "app-misc/circ-buildtime-c-1"),
-                ),
+                # RDEPEND in the other. However, it is not respected because
+                # it would result in a temporarily broken RDEPEND, so we instead
+                # rely on satisfied installed build-time dependencies.
+                # merge_order_assertions=(
+                #    ("app-misc/circ-buildtime-a-1", "app-misc/circ-buildtime-c-1"),
+                # ),
                 mergelist=[
```

Comment:
I probably won't review this tonight because I want to dig up whatever bug we added that original test case for, just to make sure the change preserves the intent too. It probably does but I'm curious as to what it was.

Reply:
Seems this particular assertion was added as an xfail in 7dc4131eca73aee2a78fd5c4339cb41760f7a54f, in preparation for a later patch related to https://bugs.gentoo.org/690436. So, it's a thing that I made up myself, rather than being a response to something that happened in the field.

Reply:
Actually https://bugs.gentoo.org/703676 is linked from https://github.com/gentoo/portage/pull/491.

Reply:
A test case that tests for bug 703676 and has the cycle shown in https://bugs.gentoo.org/703676#c1 would be good to have.

Reply:
Added RebuildGhostscriptTestCase, and it passes.

Reply:
Thank you for doing the digging! I was distracted with https://bugs.gentoo.org/917496 and didn't have a chance yet. Looks good.

Reply:
By the way, I wonder if we should at some point think about some special `--debug` mode which dumps the depgraph in the form of (a large) `ResolverPlaygroundTestCase` input for playing around with.

Reply:
Yeah, I suppose we could save state to pickle or json file and use that to populate a ResolverPlayground and ResolverPlaygroundTestCase.